test_conformance/basic/test_work_item_functions.cpp

```diff
@@ -35,146 +39,500 @@ struct work_item_data
+                log_error("ERROR: get_local_id(%d) did not return "
+                          "proper value for the argument out of range "
+                          "(expected 0, got %d)\n",
+                          (int)dim + 1, (int)testData[q].localID[0]);
+                return -1;
+            }
+            if (testData[q].numGroups[0] != 1)
+            {
+                log_error("ERROR: get_num_groups(%d) did not return "
+                          "proper value for the argument out of range "
+                          "(expected 1, got %d)\n",
+                          (int)dim + 1, (int)testData[q].numGroups[0]);
+                return -1;
+            }
+            if (testData[q].groupID[0] != 0)
+            {
+                log_error("ERROR: get_group_id(%d) did not return "
+                          "proper value for the argument out of range "
+                          "(expected 0, got %d)\n",
+                          (int)dim + 1, (int)testData[q].groupID[0]);
+                return -1;
+            }
+        }
+
+        const Version version = get_device_cl_version(device);
+        if (version >= Version(2, 0))
+        {
+            for (size_t q = 0; q < threads_to_verify; q++)
+            {
+                if (testData[q].globalOffset[0] != 0)
+                {
+                    log_error("ERROR: get_global_offset() did not return "
+                              "proper value "
+                              "for %d dimensions (expected %d, got %d)\n",
+                              (int)dim, (int)dim, (int)testData[q].workDim);
+                    return -1;
+                }
+                if (testData[q].enqueuedLocalSize[0] != 1)
+                {
+                    log_error(
+                        "ERROR: get_enqueued_local_size(%d) did not return "
+                        "proper value for the argument out of range "
+                        "(expected 1, got %d)\n",
+                        (int)dim + 1, (int)testData[q].globalSize[0]);
+                    return -1;
+                }
+            }
+        }
+        else if (version >= Version(1, 1))
+        {
+            for (size_t q = 0; q < threads_to_verify; q++)
+            {
+                if (testData[q].globalOffset[0] != 0)
+                {
+                    log_error("ERROR: get_global_offset() did not return "
+                              "proper value "
+                              "for %d dimensions (expected %d, got %d)\n",
+                              (int)dim, (int)dim, (int)testData[q].workDim);
+                    return -1;
+                }
+            }
+        }
+
+        return CL_SUCCESS;
+    }
+
+    cl_int Run() override
+    {
+        cl_int error = SetUp(kernel_source);
+        test_error(error, "SetUp failed");
+
+        cl_int dim_param = 4;
+        error = clSetKernelArg(kernel, 1, sizeof(cl_int), &dim_param);
+        test_error(error, "Unable to set kernel arg");
```

Comment:
I think this means that these tests are testing for `dimidx` equal to four exclusively, is that correct?

If so, this is better than nothing, but it means we aren't really checking for proper out-of-range behavior at `dimidx` equal to two for 1D or 2D ranges, or for `dimidx` equal to one for 1D ranges.

Reply:
Moved to the loop and set to number dimensions + 1. If you think this is still not acceptable I can follow your idea with new approach for the test - number of change requests is already large so if you don't mind I will hold at this point waiting for more feedback.